### ISSUE TITLE
some recommendations for distance sensor

### ISSUE THREAD

**Issue by eried** [dndubins/QuickStats#6]
awesome lib! I am using it for a distance sensor, I wonder if I can get some advice:

1) distance goes from the distance +/- 2, i.e. imagine distance is 50, values would be 49, 50, 51... and 5% of erratic values like 5 and 784 (max limit)

mode with 0.001 gives a quite steady value, but I am afraid of if I get 0, should I add a "mean" after mode just in case? or it would be better to use gmdn? any recommendation for the third parameter considering my numbers?

**Comment by dndubins** [comment]
Dear Erwin,<p>
Thanks for trying out the library! There are certainly different approaches to filtering. Mode can help, but one of my favourite approaches when you are having issues like you are having is rejecting values outside of the expected range. If 784 is your max limit, this could look like this:

```
int n=100;                      // number of readings
float reading=0.0;          // to hold voltage reading
float total=0.0;              // to calculate average
int duds=0;                   // to hold # dud readings
for(int i=0;i<n;i++){      // take n readings
  reading=analogRead(A1)*5.0/1023.0; //get reading in V
  // Is reading outside 10 and 770? If so, it’s a dud.
  if(reading<10.0||reading>770.0){  	    // this is where the magic happens: reject the outlier!
    duds++;                       // add 1 to duds
  }else{
    total=total+reading;     // include reading in avg
  }
}
total=total/(n-duds);        // calculate final average
```

<p>This sort of rejection approach could be applied to any of the other routines (e.g. mode). An out of spec number is thrown away before filtering.

<p>Another way to go about it is rejecting measurements that are outside ~2 standard deviations of your reading. I wrote the sketch [filterread.ino](https://github.com/dndubins/QuickStats/blob/master/examples/filterread/filterread.ino) to show an example of how this could work. Have a look! I think filterread.ino might also solve your problem.

<p>I added the gmdn function to the library more out of humour. It will give you a number for sure, and other people have picked up on it, but it came out of a [comic](https://www.explainxkcd.com/wiki/index.php?title=2435:_Geothmetic_Meandian&c-nested) from Piled Higher and Deeper. The humour I think is in its over-complexity.
